database/src/connection_pool.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright 2018 Bitwise IO
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate r2d2;
+extern crate r2d2_diesel;
+use diesel;
+use diesel::pg::PgConnection;
+use diesel::prelude::*;
+use models::*;
+use r2d2::{Pool, PooledConnection};
+use r2d2_diesel::ConnectionManager;
+use tables_schema::*;
+
+pub type Connection = r2d2::PooledConnection<ConnectionManager<PgConnection>>;
+
+pub struct ConnectionPool {
+    dsn: String,
+    pool: Pool<ConnectionManager<PgConnection>>,
+}
+
+impl ConnectionPool {
+    pub fn connect(dsn: &str) -> Result<ConnectionPool, String> {
```

Comment:
I'd consider creating an enum for connection errors. Could be useful long term handling specific errors in a `match` statement.

Something like
```
pub enum ConnectionError {
     CONNERR(String)
}
```